### DISCUSSION TITLE
options to rotate individual particle draw passes

### DISCUSSION THREAD

**Discussion by DannyDevSe** [godotengine/godot-proposals#9328] [Engine Core]
### Describe the project you are working on

making a muzzle flash and no options to rotate the induvidual passes i could prop just make them all into one texture in image editor and that is prop better but not as adjustable and found out that you can change the induvidual pass position with center offset so there should be option to also change its rotation aswell

### Describe the problem or limitation you are having in your project

not being able to rotate the draw pass

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

there already is center offset that allows you to move the individual pass would be great if there was one for rotating as well

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

n/a

### If this enhancement will not be used often, can it be worked around with a few lines of script?

n/a

### Is there a reason why this should be core and not an add-on in the asset library?

n/a

**Comment by AThousandShips** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

**Comment by Calinou** [comment]
Remember that changing Center Offset in a QuadMesh/PlaneMesh's properties will regenerate the mesh data, which is slow. It's fine to set this once in the inspector or when loading a scene, but this property should not be changed every time to avoid unnecessary performance overhead.

The same concern would apply if we implement a property that allows you rotate the mesh data while it's being generated. This means the property unfit for real-time animation by design.
